lib/std/os.zig

```diff
@@ -3237,3 +3237,20 @@ pub fn sched_yield() SchedYieldError!void {
         else => return error.SystemCannotYield,
     }
 }
+
+/// Set a socket's options.
+pub fn setsockopt(fd: fd_t, level: u32, optname: u32, opt: []const u8) !void {
+    switch (errno(system.setsockopt(fd, level, optname, opt.ptr, @intCast(socklen_t, opt.len)))) {
+        0 => {},
+        EBADF => unreachable,
+        EINVAL => unreachable,
+        EDOM => return error.TimeoutTooBig,
+        EISCONN => return error.AlreadyConnected,
+        ENOPROTOOPT => return error.InvalidProtocolOption,
+        ENOTSOCK => return error.NotSocket,
+        ENOMEM => return error.OutOfMemory,
```

Comment:
Was there a description of what this error code means? Typically when you get ENOMEM from a syscall, it means that the *kernel* is out of dedicated memory for this operation, rather than the computer's global heap memory. In the std lib we have precedent for returning `error.SystemResources` for this, such as ENOMEM from `std.os.bind`.